### DISCUSSION TITLE
Update Main Form From Relationship Manager

### DISCUSSION THREAD

**Discussion by ZevedoAFT** [filamentphp/filament#13622] [General]
I have a resource form with a relationship manager.
In the relationship manager I have a header action that, among other things, updates some fields of the main resource.
How can I refresh the main form after the action completes?

Thank you

**Comment by leandrocfe** [comment]
EditPage or ViewPage
```php
#[On('update-form')]
public function updateForm()
{ 
    $this->refreshFormData([
         'your_field',
    ]);
}
```

you can use in the action
```php
->after(function() {
     $this->dispatch('update-form');
})
```

**Comment by ZevedoAFT** [comment]
Perfect.
Thank you
